public/app/features/admin/ServerStats.tsx

```diff
@@ -13,13 +13,15 @@ export const ServerStats = () => {
   const styles = useStyles2(getStyles);
 
   useEffect(() => {
-    getServerStats().then((stats) => {
-      setStats(stats);
-      setIsLoading(false);
-    });
+    if (contextSrv.hasAccess(AccessControlAction.ActionServerStatsRead, contextSrv.isGrafanaAdmin)) {
```

Comment:
```suggestion
    if (contextSrv.hasAccess(AccessControlAction.ActionServerStatsRead, contextSrv.isGrafanaAdmin)) {
     setIsLoading(true);
```
The loading state is true by default, but if a user doesn't have the permissions, the code toggling it false is not executed, resulting in an infinite loading indicator. One way to fix it is to set `const [isLoading, setIsLoading] = useState(false);` and trigger the loading state before the api request is issued.

Reply:
Another way would be 
```
if (!contextSrv.hasAccess(AccessControlAction.ActionServerStatsRead, contextSrv.isGrafanaAdmin)) {
     setIsLoading(false);
     return;
}
```

Reply:
Ah, good point, thanks for catching!